pkg/util/kubeobjects/internal/query/query.go

```diff
@@ -0,0 +1,201 @@
+package query
+
+import (
+	"context"
+	goerrors "errors"
+
+	"github.com/Dynatrace/dynatrace-operator/pkg/logd"
+	"github.com/pkg/errors"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type Generic[T client.Object, L client.ObjectList] struct {
+	Target      T
+	ListTarget  L
+	ToList      func(L) []T
+	IsEqual     func(T, T) bool
+	IsImmutable func(T, T) bool
+
+	KubeClient client.Client
+	KubeReader client.Reader
+	Log        logd.Logger
+}
+
+func (c Generic[T, L]) Get(ctx context.Context, objectKey client.ObjectKey) (T, error) {
+	err := c.KubeReader.Get(ctx, objectKey, c.Target)
+
+	return c.Target, err
+}
+
+func (c Generic[T, L]) Create(ctx context.Context, object T) error {
+	c.Log.Info("creating", "kind", object.GetObjectKind(), "name", object.GetName(), "namespace", object.GetNamespace())
```

Comment:
is there a way to always set the attributes, `kind`/`name`/`namespace`, for the logger in a central place so we can update it more easily in the future?

Reply:
done